src/components/TradeWidget/index.tsx

```diff
@@ -403,7 +416,27 @@ const TradeWidget: React.FC = () => {
   const validFromValue = watch(validFromId)
   const validUntilValue = watch(validUntilId)
 
-  const initialPrice = useRef(priceParam)
+  // Updating global state, on change of each respective prop
+  useEffect(() => {
+    dispatch(updateTradeState('price', priceValue))
+  }, [dispatch, priceValue])
+  useEffect(() => {
+    dispatch(updateTradeState('sellAmount', sellValue))
+  }, [dispatch, sellValue])
+  useEffect(() => {
+    dispatch(updateTradeState('sellToken', sellToken as Required<TokenDetails>))
+  }, [dispatch, sellToken])
+  useEffect(() => {
+    dispatch(updateTradeState('buyToken', receiveToken as Required<TokenDetails>))
+  }, [dispatch, receiveToken])
+  useEffect(() => {
+    dispatch(updateTradeState('validFrom', validFromValue))
+  }, [dispatch, validFromValue])
+  useEffect(() => {
+    dispatch(updateTradeState('validUntil', validUntilValue))
+  }, [dispatch, validUntilValue])
```

Comment:
```suggestion
  useEffect(() => {
    dispatch(updateTradeState({
      price: priceValue,
      sellAmount: sellValue,
      sellToken,
      buyToken: receiveToken,
      validFrom: validFromValue,
      validUntil: validUntilValue
    }))
  }, [dispatch, priceValue, sellValue, sellToken, receiveToken, validFromValue, validUntilValue])
```
if you change `updateTradeState` to accept `Partial<TradeState>`